File: java/src/test/java/app/config/TargetEnvResolver.java
package app.config;

public class TargetEnvResolver {

  public record EnvInfo(String pageUrl, String baseUrl, String usersFeederFile) {}

  public static EnvInfo resolveEnvironmentInfo(String targetEnv) {
    return switch (targetEnv) {
      case "PROD" ->
          new EnvInfo(
              "https://ecomm.gatling.io", "https://api-ecomm.gatling.io", "users_dev1.json");
      default ->
          new EnvInfo(
              "https://ecomm.gatling.io", "https://api-ecomm.gatling.io", "users_dev1.json");
    };
  }
}

File: scala/src/test/scala/app/config/TargetEnvResolver.scala
package app.config

object TargetEnvResolver {

  case class EnvInfo(pageUrl: String, baseUrl: String, usersFeederFile: String)

  def resolveEnvironmentInfo(targetEnv: String): EnvInfo = targetEnv match {
    case "PROD" =>
      EnvInfo("https://ecomm.gatling.io", "https://api-ecomm.gatling.io", "users_dev1.json")
    case _ =>
      EnvInfo("https://ecomm.gatling.io", "https://api-ecomm.gatling.io", "users_dev1.json")
  }
}
